File: OrangeJuice.Server.Core.FSharp/Configuration/AuthOptionsFactory.fs
namespace OrangeJuice.Server.FSharp.Configuration

open OrangeJuice.Server.Configuration

type AuthOptionsFactory(configurationProvider : IConfigurationProvider) = 
    interface Factory.IFactory<AuthOptions> with
        member this.Create() : AuthOptions = 
            new AuthOptions(Audience = configurationProvider.GetValue("auth0:Audience"),
                            CertificateKey = configurationProvider.GetValue("auth0:CertificateKey"), 
                            Issuer = configurationProvider.GetValue("auth0:Issuer"))

File: OrangeJuice.Server.Core/Configuration/AuthOptionsFactory.cs
namespace OrangeJuice.Server.Configuration
{
	public sealed class AuthOptionsFactory : Factory.IFactory<AuthOptions>
	{
		private readonly IConfigurationProvider _configurationProvider;

		public AuthOptionsFactory(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public AuthOptions Create()
		{
			return new AuthOptions
			{
				Audience = _configurationProvider.GetValue("auth0:Audience"),
				CertificateKey = _configurationProvider.GetValue("auth0:CertificateKey"),
				Issuer = _configurationProvider.GetValue("auth0:Issuer")
			};
		}
	}
}